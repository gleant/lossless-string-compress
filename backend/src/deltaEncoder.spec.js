const { compress, decompress } = require("./deltaEncoder");

describe("Delta encoder tests", () => {
  describe("compress", () => {
    it("should compress string", () => {
      const stringToCompress = `myxa
myxophyta
myxopod
nab
nabbed
nabbing
nabit
nabk
nabob
nacarat
nacelle`;

      const result = compress(stringToCompress);

      expect(result).toMatchSnapshot();
    });
  });

  describe("decompress", () => {
    it("should decompress string", () => {
      const stringToDecompress = `0 myxa
3 ophyta
5 od
0 nab
3 bed
4 ing
3 it
3 k
3 ob
2 carat
3 elle`;

      const result = decompress(stringToDecompress);
      expect(result).toMatchSnapshot();
    });
  });
});
