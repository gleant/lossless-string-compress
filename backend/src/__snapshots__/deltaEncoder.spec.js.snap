// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delta encoder tests compress should compress string 1`] = `
"0 myxa
3 ophyta
5 od
0 nab
3 bed
4 ing
3 it
3 k
3 ob
2 carat
3 elle
"
`;

exports[`Delta encoder tests decompress should decompress string 1`] = `
"myxa
myxophyta
myxopod
nab
nabbed
nabbing
nabit
nabk
nabob
nacarat
nacelle
"
`;
